### PR TITLE
General Cypress Functional Test Maintenance

### PR DESCRIPTION
**Pull request by jujaga**
# Description

This PR performs general maintenance to our Cypress test suite. This was needed as the way we load forms, especially the introduction of the options endpoints, was updated.

- Bump Cypress to 9.1.0
- Add date-fns to handle date format generation
- Fix broken kitchen-sink test suite

## Types of changes

Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

If you have a previous version of Cypress installed locally, please make sure to `npm run purge` and then `npm ci` to get the latest Cypress updates.

### CODE REVIEW

**Comment by bcgov-nr-csst** [comment]
PR Deployed at: https://chefs-dev.apps.silver.devops.gov.bc.ca/pr-259

File: tests/functional/cypress/specs/kitchen-sink.spec.js [hunk 1]
```diff
@@ -112,12 +111,12 @@ function helperFormFields() {
 
   // checkboxGroup1
   cy.get('#eoqkh1').contains('label', 'Checkbox Group 1');
-  cy.get('#eoqkh1-check1').click();
-  cy.get('#eoqkh1-check3').click();
+  cy.get('[id*="eoqkh1--check1"]').click();
```
Comment (loneil) [inline tests/functional/cypress/specs/kitchen-sink.spec.js:114]: 👍

**Review by loneil — APPROVED** [review]
Code changes look good

**Review by TimCsaky — APPROVED** [review]
tests passed locally. looks good!

**Pull request merged**